src/verkle/mod.rs

```diff
@@ -0,0 +1,297 @@
+    (2..s.len())
+        .step_by(2)
+        .map(|i| u8::from_str_radix(&s[i..i + 2], 16))
+        .collect()
+}
+
+fn construct_headers() -> HeaderMap {
+    let mut headers = HeaderMap::new();
+    headers.insert(USER_AGENT, HeaderValue::from_static("test optimal batch size1"));
+    headers.insert(CONTENT_TYPE, HeaderValue::from_static("application/json; charset=utf-8"));
+    headers
+}
+
+// ! rlp string starts with 0x
+pub fn decode_block(rlp: String) -> Result<VerkleBlock, DecoderError> {
+    let serialized: Vec<u8> = decode_hex(&rlp).unwrap();
+    let block: VerkleBlock = decode(&serialized)?;
+
+    Ok(block)
+}
+
+pub fn print_block_info(block: &VerkleBlock) {
+    println!(
+            "Block info:\n- parent hash: {}\n- storage root: {}\n- block number: {}\n",
+            hex::encode(block.header.parent_hash.clone()),
+            hex::encode(block.header.storage_root.clone()),
+            hex::encode(block.header.number.clone())
+        );
+    let keys = block.header.keyvals.keys.clone();
+    println!("Kkey-vals: ");
+    for (indx, key) in keys.iter().enumerate() {
+        match block.header.keyvals.values[indx] {
+            Some(ref val) => println!("\t{} => {}", hex::encode(key), hex::encode(val)),
+            None => println!("\t{} is absent", hex::encode(key)),
+        }
+    }
+}
+
+pub async fn get_rlp(block_number: u64) -> Result<String, reqwest::Error> {
+    let client = Client::new();
+    let node_end_point: String = "https://rpc.condrieu.ethdevops.io/".to_owned();
+    let arg = format!(r#"{{"jsonrpc":"2.0","method":"debug_getBlockRlp","params":[{}],"id":"1"}}"#, block_number);
+
+    let res = client.post(&node_end_point)
+        .body(arg)
+        .headers(construct_headers())
+        .send()
+        .await?;
+
+    let block_rlp: ResponseTest = res.json().await?;
+
+    Ok(block_rlp.result)
+}
+
+pub fn verification(block: VerkleBlock, parent_root: String) -> Result<verkle_trie::proof::UpdateHint, anyhow::Error> {
+    let parent_root = hex::decode(parent_root).unwrap();
+    let root: EdwardsProjective = CanonicalDeserialize::deserialize(&parent_root[..]).unwrap();
+    let keyvals = block.header.keyvals;
+
+    let (checked, info) = block
+        .header
+        .proof
+        .verkle_proof
+        .check(keyvals.keys, keyvals.values, root);
+
+    match checked {
+        true => {
+            println!("Good verification");
+            Ok(info.unwrap())
+        },
+        false => {
+            println!("Bad verification");
+            Err(anyhow::anyhow!("Verification didn't work out"))
+        }
+    }
+}
+
+#[allow(dead_code)]
+pub async fn save_rlp(rlp: String, num: u64) -> Result<(), std::io::Error> {
+    use std::fs::File;
+    use std::io::prelude::*;
+
+    let mut file = File::create(format!("condrieu.block{}.rlp", num))?;
+    file.write_all(&decode_hex(&rlp).unwrap())?;
+
+    Ok(())
+}
+
+
+// Usefull, but !excess!
+
+// fn build_tree(keys, values) {
```

Comment:
you can use git to store unused but useful code. just commit it, and then delete it in another commit, and git will remember it, so you can always refer to it later